packages/matchbox/src/components/Popover/Popover.js

```diff
@@ -1,159 +1,152 @@
-        <WindowEvent event='click' handler={this.handleOutsideClick} />
-        <WindowEvent event='keydown' handler={this.handleEsc} />
+      <PopoverContent open={shouldBeOpen} popoverRef={popoverRef} {...rest}>
         {children}
       </PopoverContent>
     );
   }
 
-  renderActivator = ({ activatorRef }) => {
-    const {
-      trigger,
-      wrapper: WrapperComponent = 'span'
-    } = this.props;
-
-    const assignRefs = (node) => {
-      this.activator = node;
-      activatorRef(node);
-    };
-
+  // Renders popover trigger
+  function renderActivator({ activatorRef: forwardedRef }) {
     return (
-      <WrapperComponent
-        className={styles.Activator}
-        onClick={this.handleTrigger}
-        ref={assignRefs}>
-        {trigger}
-      </WrapperComponent>
+      <Box
+        as={Wrapper}
+        // Inline block is required to measure and set height correctly
+        display={Wrapper === 'span' ? 'inline-block' : null}
+        position="relative"
+        onClick={handleTrigger}
+        ref={activatorRef}
+      >
+        <Box
+          as={Wrapper}
+          // Inline block is required to measure and set height correctly
+          display={Wrapper === 'span' ? 'inline-block' : null}
+          width="100%"
+          height="100%"
+          ref={forwardedRef} // Component is duplicated to handle two refs
+        >
+          {trigger}
+        </Box>
+      </Box>
     );
   }
 
-  render() {
-    const { fixed, open: controlledOpen } = this.props;
-    const shouldBeOpen = controlledOpen || this.state.open;
-
-    return (
-      <PopoverOverlay
-        fixed={fixed}
-        open={shouldBeOpen}
-        portalId={this.props.portalId}
-        renderActivator={this.renderActivator}
-        renderPopover={this.renderPopover} />
-    );
-  }
+  return (
+    <PopoverOverlay
+      as={Wrapper}
+      id={id}
+      open={shouldBeOpen}
+      renderActivator={renderActivator}
+      renderPopover={renderPopover}
+      activatorRef={activatorRef}
+    />
+  );
 }
 
+Popover.displayName = 'Popover';
+Popover.propTypes = {
+  id: PropTypes.string,
+  /**
+   * A React component to will trigger the popover
+   * Click event is handled for you if this component is uncontrolled.
+   */
+  trigger: PropTypes.node,
+  /**
+   * Adds a padding to the Popover
+   */
+  sectioned: deprecate(PropTypes.bool, 'Use padding system props instead'),
+  /**
+   * Opens the popover.
+   * By default, open state is handled automatically. Passing this value in will turn this into a controlled component.
+   */
+  open: PropTypes.bool,
+  left: PropTypes.bool,
+  right: PropTypes.bool,
+  top: PropTypes.bool,
+  bottom: PropTypes.bool,
+  /**
+   * Callback function that is called when clicking outside the popover, or hitting escape.
+   */
+  onClose: PropTypes.func,
+  /**
+   * Popover Content
+   */
+  children: PropTypes.node,
+  as: PropTypes.oneOf(['div', 'span']),
+  wrapper: PropTypes.oneOf(['div', 'span']),
```

Comment:
`as` and `wrapper` here are the same thing, but figured I'd support `as` since that seems to be the convention these days – Tooltip is also using the same prop name.

Not sure about restricting this prop to div and span only

Reply:
I think we will find out.